updating readme, updating 1_get_files, adding scripts to process reference data and run instaprism

graceakatsu opened the pull request:
- updated README with required reference data files
- created 3_prepare_reference_data.R which successfully processes and combines scRNAseq from HGSOC and snRNAseq of adipose tissue

graceakatsu committed:
Update README.md

updating readme to include necessary sc data for reference

graceakatsu committed:
Update README.md

updating readme for files necessary for reference data preparation (scrna seq )

graceakatsu committed:
Add files via upload

Work in progress script to create the reference data probability matrix to input into InstaPrism.

graceakatsu committed:
Update 3_prepare_reference_data.R

still in progress, but done with the scRNAseq processing code

graceakatsu committed:
Update README.md

adding adipose tissue snRNAseq data

graceakatsu committed:
Update 3_prepare_reference_data.R

most up to date draft of this script to use in Alpine

graceakatsu committed:
creating instaprism input files folder

graceakatsu committed:
Delete instaprism input files/file.txt

graceakatsu committed:
Add files via upload

graceakatsu committed:
Update 3_prepare_reference_data.R

adding 2 lines of code to write the row and column names of the expr matrix as separate csv's

graceakatsu committed:
Delete scripts/X_run_bayesprism.R

graceakatsu committed:
Add files via upload

file to run instaprism

graceakatsu committed:
Update 1_get_data.R - changing Mayo processing

Mayo microarray dataset is unlike the other microarray datasets in that its raw data is expressed as a log10 of ratio of signal, rather than log2. (https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE74357) 

Have updated this script to process the Maya dataset as 10^(...) rather than 2^(...).

graceakatsu committed:
tiny update to 4_run_instaprism.R

saving the final output files with ".csv" extension

graceakatsu committed:
adding with and without adipocyte versions to 4_run_instaprism.R

running instaprism both with and without adipocytes for the purposes of comparison

graceakatsu committed:
adding script which visualizes instaprism outputs

creates 100% stacked bar charts to show cell type proportions (similar to Davidson et al 2024, fig 3)
also creates centered pca plots with k-means clustering (similar to Davidson et al 2024, fig 1)

graceakatsu committed:
update 2_get_clustering.R

changing so that the clustering is performed on a log10 transformed version of all datasets (as done in Davidson 2024 - https://github.com/greenelab/hgsc_characterization/blob/master/figure_notebooks/compare_centroids.Rmd). Prior code was running clustering on the _filtered_transformed versions of all datasets - this is in log10 for rnaseq datasets and in pseudo-counts for microarray datasets.

graceakatsu committed:
Update 2_get_clustering.R - undoing commit 67b85b11b3cdb4af1ea9976d9beb422e9167fa44!

ivichadriana commented on README.md:
for the readme file: can you make the subsections be easier to read? like order the # or ## or ### or ####  to represent subsections. example:
# title
## subtitle
### subtitle's subentry

graceakatsu replied:
done in updated readme!

ivichadriana commented on scripts/1_get_data.R:
can you add a short description at the top explaning what the script does, input and output

graceakatsu replied:
done, for all scripts!

ivichadriana commented on scripts/2_get_clustering.R:
```diff
@@ -1,182 +1,284 @@
-# matching original approach (rows=samples for k-means).
-###############################################
+#########################################
+# 1_get_data.R
```
this is from the other file right?

graceakatsu replied:
fixed the file path reproducibility issue, for all scripts

ivichadriana commented on scripts/2_get_clustering.R:
```diff
@@ -1,182 +1,284 @@
-# ======================
 # Set the working directory
-setwd("/Users/ivicha/Documents/deconvolution_adipocytes")
+setwd("/Users/grace/Library/CloudStorage/OneDrive-TheUniversityofColoradoDenver/greene_lab_projects/hgsoc_adipocyte_deconvolution")
```
maybe we can find a better way to make this paths be relative so people are able to directly run the script without changing code

graceakatsu replied:
fixed the file path reproducibility issue, for all scripts

ivichadriana commented on scripts/2_get_clustering.R:
```diff
@@ -1,182 +1,284 @@
+)
 
-message("Done! Clustering results written to ", final_out)
+message("All datasets processed with both asImported & transformed versions!")
```
typo here

graceakatsu replied:
fixed!

ivichadriana commented on scripts/3_prepare_reference_data.R:
```diff
@@ -0,0 +1,382 @@
+gc()
+
+# # Set the working directory
+# setwd("projects/gakatsu@xsede.org/hgsoc_adipocyte_deconvolution")
```
the same woith the paths here.. is there a way in R to make the paths be relative to the script's path?

graceakatsu replied:
fixed the file path reproducibility issue, for all scripts

ivichadriana commented on scripts/3_prepare_reference_data.R:
```diff
@@ -0,0 +1,382 @@
+gc()
+
+# Then we will create combined key that aligns cell barcodes with cell type
+all_sc_overlap_cells_labeled <- rbind(rep1_sc_overlap_cells_labeled,
```
i tihnk the adipose data might need some preprocessing (filtering) beforehand? like removing cells with expressing in less than X genes and genes not expressed in less than X cells?

graceakatsu replied:
preprocessing added to script!

ivichadriana commented on scripts/4_run_instaprism.R:
```diff
@@ -0,0 +1,120 @@
+
+# Define import and export directories
+import_dir <- file.path("/projects/gakatsu@xsede.org/hgsoc_adipocyte_deconvolution")
+export_dir <- file.path("/projects/gakatsu@xsede.org/hgsoc_adipocyte_deconvolution/instaprism_output")
```
same with these paths... there must be a better way i just have never figured it out ahaha

graceakatsu replied:
fixed the file path reproducibility issue, for all scripts

ivichadriana commented on scripts/5_visualize_instaprism_outputs.R:
```diff
@@ -0,0 +1,338 @@
+
+
+
+
```
empty space here at the bottom. also, are you doing dimensionality reduction of the proportions here?

graceakatsu replied:
removed the empty space. and no, I am not doing any dimensionality reduction as the proportions are all between 0 and 100%, so only 2 orders of magnitude different max between datapoints. I am happy to discuss though if you think I should be doing that!

ivichadriana replied:
okay we can talk about it:)

ivichadriana commented on README.md:
also we have enough files now to need some guideline o what to run in what order to replicate results. can you add a little list saying what each script does in the readme, saying they must be run in X order

graceakatsu replied:
thank you, done!

ivichadriana commented:
added some minor comments here and there. i am mostly worried about the paths for reproducibility.

ivichadriana reviewed (commented):
adding review again because i changed the rule to need a PR which i had forgotten before

graceakatsu committed:
Delete old renv directory

graceakatsu committed:
Delete old renv lockfile

graceakatsu committed:
Update README.md

adding introduction, more thorough description of scripts and file set-up

graceakatsu committed:
Update 1_get_data.R

adding more thorough descriptions, removing Mayo dataset, adding TCGA bulk dataset (in addition to TCGA microarray)

graceakatsu committed:
Update 2_get_clustering.R

adding more thorough descriptions

graceakatsu committed:
Update 3_prepare_reference_data.R

adding more thorough descriptions, and the script now performs pre-processing on adipocyte data(removal of samples with many mitochondrial gene reads, removing low quality nuclei/empty droplets/multiplets).

graceakatsu committed:
Update 4_run_instaprism.R

adding more thorough descriptions and also saving the instaprism objects as files prior to running instaprism

graceakatsu committed:
Update 5_visualize_instaprism_outputs.R

quite significant changes. this script now generates 3 kinds of plots - 100% stacked bar charts for total cell proportions per dataset, 100% stacked bar charts for cell proportions per sample in each dataset, and bar charts showing absolute change in cell type proportions per dataset.

graceakatsu committed:
Update README.md

adding bullet points to file names for readability, and updating directory screenshot to specify project name

graceakatsu committed:
adding new renv lockfile

ivichadriana reviewed (approved):
approving

graceakatsu committed:
Update 1_get_data.R

fixing to only include HGSOC and only include samples in common between TCGA microarray + RNAseq

graceakatsu committed:
Update README.md

changing *** to ... in order to prevent bolding of the text

graceakatsu committed:
Update README.md

in order to use only labeled adipocytes for the sn data. updated the description of 3_prepare_reference_data.R to indicate filtering for adipocytes only, and also added the information about the required metadata file.

graceakatsu committed:
Update 3_prepare_reference_data.R

in order to use only labeled adipocytes for the sn data.

graceakatsu committed:
Update README.md

adding information about the female sn adipose samples that i had not included prior

graceakatsu committed:
Update 3_prepare_reference_data.R

adding the sn adipose data that was not previously used. also making the sn processing script more robust to different #s of samples

graceakatsu committed:
Update 3_prepare_reference_data.R

fixing tiny typo in an object name

graceakatsu committed:
Update README.md

adding the new file sources for the TCGA bulk dataset

graceakatsu committed:
Update 1_get_data.R

updating for new TCGA bulk data

graceakatsu committed:
deleting old lockfile

graceakatsu committed:
adding new lockfile

graceakatsu committed:
Update 1_get_data.R

fixing the gene names of new tcga bulk data

graceakatsu committed:
Update 1_get_data.R

fixing the gene names of tcga bulk so that they are in hgnc format, and dealing with duplicates

graceakatsu committed:
Update 1_get_data.R

adding a line to deal with NA values in tcga_bulk
